### PR TITLE
Data-3395: Add GetLatestTabularData

### PR DESCRIPTION
**Pull request by gloriacai01**
We added a new Data endpoint GetLatestTabularData [here](https://github.com/viamrobotics/app/pull/6896), this PR adds the endpoint to python SDK.

### CODE REVIEW

File: src/viam/app/data_client.py
Comment (njooma) [inline src/viam/app/data_client.py:311]: Since some of the return values are optional, I think they should be moved to be after the non-optional value in the tuple

File: src/viam/app/data_client.py
Comment (njooma) [inline src/viam/app/data_client.py:333]: rearrange these return types to match the order they are returned in the tuple

**Review by njooma — CHANGES_REQUESTED** [review]
minor changes around return types, otherwise looks good!

File: src/viam/app/data_client.py
Comment (kaywux) [inline src/viam/app/data_client.py:318]: Add resource_subtype?

File: src/viam/app/data_client.py
Comment (kaywux) [inline src/viam/app/data_client.py:340]: If we get an empty GetLatestTabularDataResponse (aka data hasn't been synced yet for the data source) could we return None for the whole return tuple? I don't like how only the time_captured and time_synced fields are optional and the payload isn't, since those are associated with the payload and should only be returned when the payload is also present.

I'm not very familiar with python - so let me know if this is not super python-y!

**Review by kaywux — CHANGES_REQUESTED** [review]

File: src/viam/app/data_client.py
Comment (gloriacai01) [reply src/viam/app/data_client.py]: yes makes sense, i changed the entire tuple to be optional so we can return None

File: src/viam/app/data_client.py
Comment (purplenicole730) [inline src/viam/app/data_client.py:311]:
```suggestion
    async def get_latest_tabular_data(self, part_id: str, resource_name: str, resource_subtype: str, method_name: str) -> Optional[Tuple[datetime, datetime, Dict[str, ValueTypes]]]:
```
Just a simple whitespace thing

File: tests/mocks/services.py
Comment (purplenicole730) [inline tests/mocks/services.py:1017]: I suspect this is one too many new lines added

**Review by purplenicole730 — COMMENTED** [review]
Mostly looks good to me, but wanted to hold off approving until after the tests pass

File: src/viam/app/data_client.py
Comment (njooma) [inline src/viam/app/data_client.py:331]: Reorder this tuple to be the actual return order

**Review by njooma — CHANGES_REQUESTED** [review]

File: src/viam/app/data_client.py
Comment (kaywux) [inline src/viam/app/data_client.py:303]: Could you reorder the returned tuple to mirror the api message field ordering? Would be good to be consistent with order across SDKs. ie Tuple[datetime, datetime, Dict[str, ValueTypes]]. Here and below!

File: src/viam/app/data_client.py
Comment (kaywux) [inline src/viam/app/data_client.py:323]: "A return value of None means that data hasn't been synced yet for the data source or the most recently captured data was over a year ago"

**Review by kaywux — CHANGES_REQUESTED** [review]

**Review by njooma — APPROVED** [review]
Approved % @kaywux's comments

File: src/viam/app/data_client.py
Comment (kaywux) [inline src/viam/app/data_client.py:323]: Comment still has wrong order in return tuple!

**Review by kaywux — CHANGES_REQUESTED** [review]

**Review by kaywux — APPROVED** [review]
Thanks Gloria! LGTM

**Pull request merged**
